test/unit/plugins/rest_spec.js

```diff
@@ -675,425 +675,853 @@ describe("The Rest plugin", function () {
-					var request = new Request("PUT", "/v1/instances/" + instance.id).mime("application/json")
-					.payload({
-						ami      : instance.ami,
-						type     : instance.type,
-						uri      : null,
-						state    : "terminating",
-						revision : instance.revision
+							state    : "terminating",
+							revision : instance.revision
+						});
+						return request.inject(server);
+					})
+					.then(function (response) {
+						result = response.result;
 					});
+				});
 
-					return Bluebird.join(request.inject(server), request.inject(server));
-				})
-				.then(function (results) {
-					responses = results;
+				after(function () {
+					updateStub.restore();
 				});
 
+				it("throws a 500 error", function () {
+					expect(result.statusCode).to.equal(500);
+				});
 			});
 
-			after(function () {
-				terminateInstancesStub.restore();
+			describe("when the database fails after terminateInstances has run", function () {
+
+				var result;
+				var updateStub;
+				var terminateInstancesStub;
+				var server     = new Hapi.Server();
+				var instances  = new InstanceAdapter();
+				var awsAdapter = new AwsAdapter();
+				before(function (done) {
+
+					return instances.createInstance()
+					.then(function (instance) {
+						updateStub = Sinon.stub(instances, "updateInstance").returns(
+							Bluebird.resolve({
+								id       : instance.id,
+								ami      : instance.ami,
+								type     : instance.type,
+								state    : "terminating",
+								uri      : instance.uri,
+								revision : instance.revision + 1
+							})
+						);
+						updateStub.onCall(1).rejects(new Error("Connection to the database fails."));
+
+						//revision reflects the document is updated twice when terminateInstances() is successful.
+						terminateInstancesStub = Sinon.stub(awsAdapter, "terminateInstances", function () {
+							return Bluebird.resolve({
+								id       : instance.id,
+								ami      : instance.ami,
+								type     : instance.type,
+								state    : "terminated",
+								uri      : instance.uri,
+								revision : instance.revision + 2
+							})
+							.then(function () {
+								done();
+							});
+						});
+
+						server.connection();
+						server.registerAsync({
+							register : Rest,
+							options  : {
+								instances  : instances,
+								awsAdapter : awsAdapter
+							}
+						});
+
+						var request = new Request("PUT", "/v1/instances/" + instance.id).mime("application/json")
+						.payload({
+							ami      : instance.ami,
+							type     : instance.type,
+							uri      : instance.uri,
+							state    : "terminating",
+							revision : instance.revision
+						});
+						return request.inject(server);
+					})
+					.then(function (response) {
+						result = response.result;
+					});
+				});
+
+				after(function () {
+					terminateInstancesStub.restore();
+					updateStub.restore();
+				});
+
+				it("leaves the state unchanged(terminating)", function () {
```

Comment:
Is this the correct test result? The way I read the code, if there is an error after requesting a terminated instance, the instance should attempt to update to the "failed" state. I know that, due to the way you're using stubs and `done`, you don't actually reach that update. But, I think it may be possible to if you make some adjustments to the strategy you've developed.